analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContext.cs

```diff
@@ -121,9 +128,37 @@ public void RegisterNodeAction<TSyntaxKind>(GeneratedCodeRecognizer generatedCod
         // the decision is made on based on whether the project contains the analyzer as NuGet).
         if (context.HasMatchingScope(supportedDiagnostics)
             && context.ShouldAnalyzeTree(sourceTree, generatedCodeRecognizer)
+            && context.Options.ParseSonarLintXmlSettings() is { } sonarLintSettings
+            && IsIncluded(context.Compilation, PropertiesHelper.ReadSourceFileInclusionsProperty(sonarLintSettings, context.Compilation.Language), context.Tree.FilePath)
+            && !IsExcluded(context.Compilation, PropertiesHelper.ReadSourceFileInclusionsProperty(sonarLintSettings, context.Compilation.Language), context.Tree.FilePath)
             && LegacyIsRegisteredActionEnabled(supportedDiagnostics, context.Tree))
         {
             action(context);
         }
     }
+
+    private bool IsIncluded(Compilation compilation, string[] inclusions, string filePath) =>
+        inclusions is null
```

Comment:
If you always return an empty array if there are no inclusions/ exclusions then you will not need the null check.